packages/api/tests/endpoints/test-granules.js

```diff
@@ -3640,45 +3417,143 @@ test.serial('associateExecution (POST) returns Not Found if collectionId in payl
     knex,
   } = t.context;
   const {
+    newPgGranule,
     newDynamoGranule,
+    esRecord,
   } = await createGranuleAndFiles({
     dbClient: knex,
     esClient,
-    granuleParams: { status: 'running', execution: executionUrl },
+    executionCumulusId: executionPgRecord.cumulus_id,
+    granuleParams: {
+      beginningDateTime: new Date().toString(),
+      cmrLink: 'example.com',
+      createdAt: Date.now(),
+      duration: 1000,
+      endingDateTime: new Date().toString(),
+      error: { errorKey: 'errorValue' },
+      execution: executionUrl,
+      lastUpdateDateTime: new Date().toString(),
+      processingEndDateTime: new Date().toString(),
+      processingStartDateTime: new Date().toString(),
+      productionDateTime: new Date().toString(),
+      productVolume: '1000',
+      published: true,
+      queryFields: { queryFieldsKey: 'queryFieldsValue' },
+      status: 'completed',
+      timestamp: 1,
+      timeToArchive: 1000,
+      timeToPreprocess: 1000,
+      updatedAt: Date.now(),
+    },
   });
 
-  const fakeEsClient = {
-    update: () => {
-      throw new Error('something bad');
-    },
-    delete: () => Promise.resolve(),
+  console.log(esRecord);
+  // TODO validate granules exist
+  const newGranule = {
+    granuleId: newDynamoGranule.granuleId,
+    collectionId: newDynamoGranule.collectionId,
+    status: 'completed',
   };
 
-  const updatedGranule = {
-    ...newDynamoGranule,
+  await request(app)
+    .put(`/granules/${newDynamoGranule.granuleId}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(newGranule)
+    .expect(200); // 200 should be expected for *update*
+
+  const actualGranule = await t.context.granuleModel.get({
+    granuleId: newDynamoGranule.granuleId,
+  });
+
+  const expectedGranule = {
+    ...newGranule,
+    error: {}, // This is a default value for no execution
+    published: false, // This is a default value
+    execution: newDynamoGranule.execution, // This should not have changed
     status: 'completed',
+    timestamp: actualGranule.timestamp,
+    updatedAt: actualGranule.updatedAt,
+    createdAt: actualGranule.createdAt,
   };
 
-  const expressRequest = {
-    params: {
-      granuleName: updatedGranule.granuleId,
-    },
-    body: updatedGranule,
-    testContext: {
-      knex,
-      esClient: fakeEsClient,
-    },
+  const actualPgGranule = await t.context.granulePgModel.get(t.context.knex, {
+    cumulus_id: newPgGranule.cumulus_id,
+  });
+  const translatedActualPgGranule = await translatePostgresGranuleToApiGranule({
+    knexOrTransaction: knex,
+    granulePgRecord: actualPgGranule,
+  });
+
+  const updatedEsRecord = await t.context.esGranulesClient.get(
+    newDynamoGranule.granuleId
+  );
+
+  t.deepEqual(actualGranule, expectedGranule);
+  // Files is always returned as '[]' by translator if none exist
+  t.deepEqual(
+    { ...translatedActualPgGranule },
+    { ...expectedGranule, files: [] }
```

Comment:
If `files` is deleted from data stores, should the actual files get deleted from s3?

Reply:
No, at least not in *this* PR - none of our other granules endpoint APIs do that *except* delete granules because they were implemented under different feature requests (e.g. delete granule was implemented before modification/insertions was possible as a file/granule management feature whereas this API was built with expectations of database management only, like the workflow granule.files (they're not removed on duplicate ingest). 

I'd be happy to talk more about that in detail, I feel pretty strongly that the API should be consistent, and we should have a discussion about reingest and excess files.